### ISSUE TITLE
GRPO can not train in single batch (attached configuration)? GRPO 训练过程未报错，但模型未发生改变，训练失败

### ISSUE THREAD

**Issue by HarideP** [modelscope/ms-swift#4348]
## 问题描述 Question 

由于硬件受限，只有两个32G的显卡，训练DeepSeek-R1-Distill-Qwen-1.5B，输出最大长度为8k，但是由于显存较小，批次只能为1。所以我使用梯度累计的方法，梯度累计为32。训练程序可以正常运行，但是训练后模型参数似乎没有发生改变，训练似乎是无效的，kl也一直为0。想请教这是一个BUG还是我配置有误。训练参数信息如下：

Due to hardware limitations, there are only two 32G graphics cards. When training DeepSeek-R1-Distill-Qwen-1.5B with a maximum output length of 8k, the batch size can only be 1 because of the limited VRAM. Therefore, I used gradient accumulation with a gradient accumulation step of 32. The training program runs normally, but the model parameters do not seem to change after training, and the training appears to be ineffective. The KL divergence is always 0. I would like to ask if this is a bug or if I have configured something incorrectly. The training parameter information is as follows:

## 训练参数 Configuration

```bash
max_length=$((1024 * 1)) 
max_response_length=$((1024 * 8)) 
per_device_train_batch_size=1
per_device_eval_batch_size=4
gradient_accumulation_steps=32
output_dir=Qwen2.5-Distill-1.5B-GRPO
swift rlhf \
    --rlhf_type grpo \
    --model ${model_path} \
    --external_plugins ${external_plugins_path} \
    --reward_funcs ${reward_funcs} \
    --use_vllm true \
    --vllm_mode colocate \
    --vllm_gpu_memory_utilization 0.8 \
    --vllm_max_model_len 9216 \
    --vllm_tensor_parallel_size 2 \
    --train_type lora \
    --lora_rank 8 \
    --lora_alpha 32 \
    --target_modules all-linear \
    --torch_dtype bfloat16 \
    --dataset ${train_data_path} \
    --val_dataset ${test_data_path} \
    --dataset_shuffle false \
    --max_length ${max_length} \
    --max_completion_length ${max_response_length} \
    --num_train_epochs 1 \
    --per_device_train_batch_size ${per_device_train_batch_size} \
    --per_device_eval_batch_size ${per_device_eval_batch_size} \
    --learning_rate 1e-6 \
    --gradient_accumulation_steps ${gradient_accumulation_steps} \
    --eval_on_start false \
    --eval_steps -1 \
    --save_steps 90 \
    --save_total_limit 1 \
    --logging_steps 1 \
    --output_dir ${output_dir} \
    --warmup_steps 20 \
    --dataloader_num_workers 4 \
    --dataset_num_proc 4 \
    --num_generations 8 \
    --temperature 0.9 \
    --log_completions true \
    --deepspeed zero2_offload \
    --sleep_level 1 \
    --offload_optimizer true \
    --offload_model true \
    --gc_collect_after_offload true \
    --use_liger_kernel true \
    --report_to wandb

```

**Comment by hjh0119** [comment]
Check if reward_std during the training process is 0; if it is not 0, upgrade DeepSpeed and try again.
